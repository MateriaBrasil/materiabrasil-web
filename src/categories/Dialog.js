import React from 'react'
import { List } from 'croods'

import Dialog from 'materials/Dialog'
import renderList from './renderList'
import ToggleExpandController from './ToggleExpandController'

export default props => (
  <ToggleExpandController
    {...props}
    render={controllerProps => (
      <Dialog {...controllerProps} title="Editar informações técnicas">
        <List name="categories" render={renderList(controllerProps)} />
      </Dialog>
    )}
  />
)
