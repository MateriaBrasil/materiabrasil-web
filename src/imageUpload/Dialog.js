import React, { Component } from 'react'
import Dialog from '@material-ui/core/Dialog'
import DialogTitle from '@material-ui/core/DialogTitle'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import withMobileDialog from '@material-ui/core/withMobileDialog'
import Link from 'react-router-dom/Link'
import Button from '@material-ui/core/Button'

import closeDialog from './closeDialog'

export default withMobileDialog()(
  class extends Component {
    render() {
      const { fullScreen, disabled, onUpload, children } = this.props
      const handleCloseDialog = closeDialog(this.props)

      return (
        <Dialog
          fullScreen={fullScreen}
          open
          aria-labelledby="responsive-dialog-title"
          maxWidth="md"
          fullWidth
          onBackdropClick={handleCloseDialog}
          onEscapeKeyDown={handleCloseDialog}
        >
          <DialogTitle id="responsive-dialog-title">Alterar imagem</DialogTitle>
          <DialogContent>{children}</DialogContent>
          <DialogActions>
            <Link to="/profile" style={{ textDecoration: 'none' }}>
              <Button color="primary">Cancelar</Button>
            </Link>
            <Button disabled={disabled} color="primary" onClick={onUpload}>
              Salvar
            </Button>
          </DialogActions>
        </Dialog>
      )
    }
  },
)
