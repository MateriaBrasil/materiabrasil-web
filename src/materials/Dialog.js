import React, { Component } from 'react';
import isFunction from 'lodash/isFunction';
import Dialog from '@material-ui/core/Dialog';

import {
  DialogTitleExtended,
  DialogContentExtend,
} from './questionnairesStyles';

import withMobileDialog from '@material-ui/core/withMobileDialog';

import DialogActions from './DialogActions';
import closeModal from './closeModal';

export default withMobileDialog()(
  class extends Component {
    constructor(props) {
      super(props);
      this.onDialogSubmit = this.onDialogSubmit.bind(this);
    }

    onDialogSubmit(event) {
      const { handleSubmit, onSubmit } = this.props;
      if (isFunction(handleSubmit)) {
        return handleSubmit(onSubmit)(event);
      }
    }

    render() {
      const { title, children, fullScreen, match, history } = this.props;
      const { onCloseModel } = this.props;
      const { closeButtonName } = this.props || false;
      const { id } = match.params;
      const handleCloseModal = onCloseModel || closeModal({ history, id });

      console.log('------------------------------------');
      console.log(children);
      console.log('------------------------------------');

      return (
        <Dialog
          fullScreen={fullScreen}
          open
          aria-labelledby="responsive-dialog-title"
          onBackdropClick={handleCloseModal}
          onEscapeKeyDown={handleCloseModal}
          fullWidth
        >
          <form onSubmit={this.onDialogSubmit}>
            <DialogTitleExtended id="responsive-dialog-title">
              {title}
            </DialogTitleExtended>

            <DialogContentExtend>{children}</DialogContentExtend>
            <DialogActions {...this.props} onCloseModal={handleCloseModal} />
          </form>
        </Dialog>
      );
    }
  },
);
