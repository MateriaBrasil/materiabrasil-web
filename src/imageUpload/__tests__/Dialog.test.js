import React from 'react'
import renderer from 'react-test-renderer'

import Dialog from '../Dialog'

jest.mock('react-router-dom/Link', () => props => (
  <div {...props}>{props.children}</div>
))
jest.mock('@material-ui/core/Dialog', () => props => (
  <div {...props}>Dialog - {props.children}</div>
))

it('renders correctly', () => {
  const tree = renderer.create(<Dialog />).toJSON()
  expect(tree).toMatchSnapshot()
})
