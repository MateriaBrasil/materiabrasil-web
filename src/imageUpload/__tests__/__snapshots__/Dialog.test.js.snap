// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  aria-labelledby="responsive-dialog-title"
  fullScreen={false}
  fullWidth={true}
  maxWidth="md"
  onBackdropClick={[Function]}
  onEscapeKeyDown={[Function]}
  open={true}
>
  Dialog - 
  <div
    className="MuiDialogTitle-root-1"
    id="responsive-dialog-title"
  >
    <h2
      className="MuiTypography-root-2 MuiTypography-title-8"
    >
      Alterar imagem
    </h2>
  </div>
  <div
    className="MuiDialogContent-root-37"
  />
  <div
    className="MuiDialogActions-root-38"
  >
    <button
      className="MuiButtonBase-root-66 MuiButton-root-40 MuiButton-text-42 MuiButton-textPrimary-43 MuiButton-flat-45 MuiButton-flatPrimary-46 MuiDialogActions-action-39"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-41"
      >
        Cancelar
      </span>
      <span
        className="MuiTouchRipple-root-69"
      />
    </button>
    <button
      className="MuiButtonBase-root-66 MuiButton-root-40 MuiButton-text-42 MuiButton-textPrimary-43 MuiButton-flat-45 MuiButton-flatPrimary-46 MuiDialogActions-action-39"
      disabled={false}
      onBlur={[Function]}
      onClick={undefined}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-41"
      >
        Salvar
      </span>
      <span
        className="MuiTouchRipple-root-69"
      />
    </button>
  </div>
</div>
`;
